File: CPP/Abr_07_Strings_CPP/Qual_ASCII.cpp
#include <iostream>

using namespace std;

int main() {
  int i;

  while (true) {
    do {
       cout << "Posição, -1 para encerrar: ";
       cin >> i;
    } while ((i < -1) || (i > 255));

    if (i == -1)
        break;

    cout << i << " - " << (char)i << endl << endl;
  }

  return(0);
}

File: POO_2026_1/src/Abr_07_Strings/Qual_ASCII.java
package Abr_07_Strings;

import java.util.Scanner;

public class Qual_ASCII {

    public static void main(String[] args) {
      Scanner ler = new Scanner(System.in);
      int i;
      
      while (true) {
          
        do {
           System.out.print("Posição, -1 para encerrar: ");
           i = ler.nextInt();
        } while ((i < -1) || (i > 255));
        
        if (i == -1)
            break;
        
        System.out.printf("%d - %c\n\n", i, (char)i);        
      }
    }

}
